include/perf_watcher.h

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+#pragma once
+
+#include <linux/perf_event.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+struct PerfWatcherOptions {
+  bool is_kernel;
+  bool is_freq;
+};
+
+typedef struct PerfWatcher {
+  const char *desc;
+  uint64_t sample_type;
+  int type;
+  unsigned long config;
+  union {
+    uint64_t sample_period;
+    uint64_t sample_frequency;
+  };
+  int sample_type_id; // index into the sample types defined in this header
+  // perf_event_open configs
+  struct PerfWatcherOptions options;
+  // tracepoint configuration
+  uint8_t reg;
+  uint8_t trace_off;
+  uint8_t trace_sz;
+  const char *tracepoint_name;
+  const char *tracepoint_group;
+  // Other configs
+  bool send_pid;
+  bool send_tid;
+  int pprof_sample_idx;       // index into the SampleType in the pprof
```

Comment:
The comments are helpful thanks !
As mentioned by Nicolas, if we had c++ or an init function, we would benefit from having these init at -1